Introduces tests for the gateway class

tobsan opened the pull request:
This means that lots of tests that did the same things in each
of the gateway subclasses are being removed.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Introduces tests for the gateway class

This means that lots of tests that did the same things in each
of the gateway subclasses are being removed.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

breakreturn commented on libsoftwarecontainer/unit-test/gateway_unittest.cpp:
```diff
@@ -0,0 +1,193 @@
+        EXPECT_CALL(gw, readConfigElement(_));
+        ASSERT_TRUE(gw.setConfig(validConf));
+    }
+}
```
I'd like to see more comments explaining what the tests does.

tobsan replied:
I've added comments now.

Commit:
gateway unit tests: changes to a nice mock

With this, we don't need to expect calls to the mocked off functions
if we don't want to - and we usually don't.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
